### PR TITLE
feat(RHINENG-5479): Update API to handle Service Account type

### PR DESCRIPTION
**Pull request by FabriciaDinizRH**
# Overview

This PR is being created to address [RHINENG-5479](https://issues.redhat.com/browse/RHINENG-5479).
It adds the Identity type `ServiceAccount` and updates tests.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [x] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by kruai — APPROVED** [review]
LGTM!

File: app/auth/identity.py [hunk 1]
```diff
@@ -144,6 +153,15 @@ class SystemIdentitySchema(IdentityBaseSchema):
     system = m.fields.Nested(SystemInfoIdentitySchema, required=True)
 
 
+class ServiceAccountInfoIdentitySchema(IdentityBaseSchema):
+    client_id = m.fields.Str(required=True, validate=verify_uuid_format)
+    username = m.fields.Str(required=True, validate=m.validate.Length(min=1))
+
+
+class ServiceAccountIdentitySchema(IdentityBaseSchema):
+    service_account = m.fields.Nested(ServiceAccountInfoIdentitySchema, required=True)
```
Comment (thearifismail) [inline app/auth/identity.py:162]: Should there be a test for `m.fields.Nested(ServiceAccountInfoIdentitySchema, **required=False**)`?

File: app/auth/identity.py [hunk 2]
```diff
@@ -155,6 +164,15 @@ class SystemIdentitySchema(IdentityBaseSchema):
     system = m.fields.Nested(SystemInfoIdentitySchema, required=True)
 
 
+class ServiceAccountInfoIdentitySchema(IdentityBaseSchema):
+    client_id = m.fields.Str(required=True, validate=verify_uuid_format)
```
[the remark this answers is not in the document]
Comment (FabriciaDinizRH) [reply app/auth/identity.py:168]: yeah, and it definitelyseems to be causing problems, I`ll fix that

[the remark this answers is not in the document]
Comment (kruai) [reply app/auth/identity.py:168]: Oh, I thought it was a UUID too... must have been thinking of `rhc_client_id`.

[the remark this answers is not in the document]
Comment (vkrizan) [reply app/auth/identity.py:168]: It might be a UUID, but I do not think it is guaranteed.

**Pull request merged**